src/base/BaseTool.js

```diff
@@ -111,16 +111,35 @@ export default class BaseTool {
   /**
    * Applys the requested mixins to the class.
    *
-   * @param  {Array} mixinArray An array of mixin names (strings).
+   * @param  {Array} mixinArray An array of mixin URIs (strings).
    */
   _applyMixins (mixins) {
+    // TODO:
+    // TODO: deal with: missing namespace + type
+
+    if (typeof cToolsImport !== 'function') { // Makes testing a lot easier.
+      return;
+    }
+
     for (let i = 0; i < mixins.length; i++) {
-      const mixinName = mixins[i];
+      let mixinUri = mixins[i];
+
+      const split = mixinUri.split('/');
+
+      if (split.length === 2) {
+        // Fill in ommited type term (mixin).
+        mixinUri = `${split[0]}/mixins/${split[1]}`;
+      } else if (split.length === 1) {
+        // Assume core if no namespace given.
+        mixinUri = `core/mixins/${split[0]}`;
+      }
```

Comment:
Okay, so we're checking for...

`moduleName/mixins/mixinName` OR
`core/mixins/mixinName`

I'm wondering if it would be more valuable to have all mixins in the same namespace, and enforce no name collisions? If you're importing multiple plugins, they could override other mixins if you're using the simple string. I think it might be easier to just have them vendor prefix the mixins name itself:

- All mixins are at `mixins/mixinName`
- Vendors should adopt the following pattern when creating a mixin name
    - Vendor prefix: `ast` (for Asteris); Mixin name: `alertEverything`
    - When registering `.register('astAlertEverything')`

There's less confusion around what namespace to use, all namespaces are now smaller, and mixin namespace collision due to modules becomes less likely? FWIW, vendor prefixing like that is a popular concept in WebComponents.

Reply:
We can easily prefix based on the plugin/vendor name. My main motivation for this approach is in the other comment.

Perhaps we can scour and build an object upon request if we are clever about the prefixing. Will discuss strategies in a while :+1: :).